### PR TITLE
DOC: Improved docs for jnp.fix and jnp.trunc

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
Part of #21461

Rendered preview:
- https://jax--23233.org.readthedocs.build/en/23233/_autosummary/jax.numpy.fix.html
- https://jax--23233.org.readthedocs.build/en/23233/_autosummary/jax.numpy.trunc.html

### CODE REVIEW

File: jax/_src/numpy/lax_numpy.py [hunk 1]
```diff
@@ -465,9 +465,37 @@ def result_type(*args: Any) -> DType:
...
+  """Return the element-wise truncated value of the input.
+
+  JAX implementation of :func:`numpy.trunc`.
+
+  Args:
+    x: input array or scalar.
+
+  Returns:
+    An array containing the truncated values of ``x``. The truncated values of the
+    elements of ``x`` are closer to ``0``, than the elements themselves.
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:479]: This description doesn't make clear that the truncated values are integers, or that they're the closest integers to the values of ``x``.

Also, we should specify the dtype of the output (are the outputs integer-typed for floating-point input?)

File: jax/_src/numpy/lax_numpy.py [hunk 2]
```diff
@@ -2558,9 +2586,38 @@ def _round_float(x: ArrayLike) -> Array:
 round_ = round
 
 
-@util.implements(np.fix, skip_params=['out'])
 @jit
 def fix(x: ArrayLike, out: None = None) -> Array:
+  """Round input to the nearest integer towards zero.
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:2591]: This is a much better description than the one in `trunc` – I read it and I know immediately what the function does!

File: jax/_src/numpy/lax_numpy.py [hunk 3]
```diff
@@ -2558,9 +2586,38 @@ def _round_float(x: ArrayLike) -> Array:
...
+
+  JAX implementation of :func:`numpy.fix`.
+
+  Args:
+    x: input array.
+    out: unused by JAX.
+
+  Returns:
+    An array containing the rounded values of ``x``.
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:2600]: An array with the same shape and dtype as ``x``

Comment (rajasekharporeddy) [reply jax/_src/numpy/lax_numpy.py]: Updated

File: jax/_src/numpy/lax_numpy.py (hunk 1 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/lax_numpy.py]: Updated the description

File: jax/_src/numpy/lax_numpy.py [hunk 4]
```diff
@@ -467,16 +467,16 @@ def result_type(*args: Any) -> DType:
 
 @jit
 def trunc(x: ArrayLike) -> Array:
-  """Return the element-wise truncated value of the input.
+  """Return truncated value of the input to the nearest integer towards zero.
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:470]: I think `Round input to the nearest integer towards zero.` would be far more clear here.

File: jax/_src/numpy/lax_numpy.py [hunk 5]
```diff
@@ -2597,11 +2597,11 @@ def fix(x: ArrayLike, out: None = None) -> Array:
     out: unused by JAX.
 
   Returns:
-    An array containing the rounded values of ``x``.
+    An array with same shape and dtype as ``x``.
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:2600]:
```suggestion
    An array with same shape and dtype as ``x`` containing the rounded values.
```

File: jax/_src/numpy/lax_numpy.py [hunk 6]
```diff
@@ -467,16 +467,16 @@ def result_type(*args: Any) -> DType:
...
   JAX implementation of :func:`numpy.trunc`.
 
   Args:
     x: input array or scalar.
 
   Returns:
-    An array containing the truncated values of ``x``. The truncated values of the
-    elements of ``x`` are closer to ``0``, than the elements themselves.
+    An array with same shape and dtype as ``x``, containing the values of
+    ``x`` truncated to the nearest integer towards ``0``.
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:479]: You could use the same `Returns` description here as in `fix` below, it's much more clear and concise.

File: jax/_src/numpy/lax_numpy.py (hunk 4 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/lax_numpy.py]: Modified as suggested

File: jax/_src/numpy/lax_numpy.py (hunk 5 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/lax_numpy.py]: Updated

File: jax/_src/numpy/lax_numpy.py [hunk 7]
```diff
@@ -2558,9 +2585,38 @@ def _round_float(x: ArrayLike) -> Array:
...
+
+  Args:
+    x: input array.
+    out: unused by JAX.
+
+  Returns:
+    An array with same shape and dtype as ``x`` containing the rounded values.
+
+  See also:
+    - :func:`jax.numpy.trunc`: Returns the truncated value of the input to nearest
```
[the remark this answers is not in the document]
Comment (rajasekharporeddy) [reply jax/_src/numpy/lax_numpy.py]: Updated.

**Review by jakevdp — APPROVED** [review]
Thanks!

**Pull request merged**